File: Game_of_Life.java
public class Solution {
    private int[] dx = {0, 1, 0, -1, 1, -1, -1, 1};
    private int[] dy = {1, 0, -1, 0, 1, 1, -1, -1};
    public void gameOfLife(int[][] board) {
        for(int i=0; i<board.length; i++) {
            for(int j=0; j<board[i].length; j++) {
                int count = 0;
                for(int k=0; k<dx.length; k++) {
                    int xx = i + dx[k];
                    int yy = j + dy[k];
                    count += status(board, xx, yy);
                }
                if(count == 3) board[i][j] |= 2;
                else if (count >=2 && count <= 3 && status(board, i, j) > 0) {
                    board[i][j] |= 2;
                }
            }
        }
        for(int i=0; i<board.length; i++) {
            for(int j=0; j<board[i].length; j++) {
                board[i][j] = board[i][j] >> 1;
            }
        }
    }
    
    private int status(int[][] board, int x, int y) {
        if(x >=0 && x < board.length && y >=0 && y<board[0].length) return 1 & board[x][y];
        return 0;
    }
}

File: Swift/game-of-life.swift
/**
 * https://leetcode.com/problems/game-of-life/
 * 
 * 
 */ 
// Date: Wed Dec 30 09:10:03 PST 2020
class Solution {
	
	func gameOfLife(_ board: inout [[Int]]) {
		
		guard board.count > 0 else { return }
		guard board[0].count > 0 else { return }
		
		for i in 0..<board.count {
			for j in 0..<board[0].count {
				
				// cell is dead
				if board[i][j] == 0 {
					if getLivesCount(board, i, j) == 3 {
						board[i][j] = 3 // dead -> alive
					}
				}
				
				// cell is alive
				if board[i][j] == 1 {
					if getLivesCount(board, i, j) < 2 || getLivesCount(board, i, j) > 3 {
						board[i][j] = 2 // alive -> dead
					}
				}
			}
		}
		
		for i in 0..<board.count {
			for j in 0..<board[0].count {
				board[i][j] %= 2
			}
		}
	}
	
	private func getLivesCount(_ board: [[Int]], _ i: Int, _ j: Int) -> Int {
		let dimensions: [[Int]] = [
			[-1, -1], [-1, 0], [-1, 1], [0, -1], [0, 1], [1, -1], [1, 0], [1, 1]]
		var lives = 0
		
		for d in dimensions {
			let x = d[0] + i
			let y = d[1] + j
			guard x >= 0, x < board.count, y >= 0, y < board[0].count else {
				continue
			}
			if board[x][y] == 1 || board[x][y] == 2 {
				lives += 1
			}
		}
		return lives
	}
}